test/apps/iodemo/io_demo.cc

```diff
@@ -1410,6 +1409,44 @@ class DemoClient : public P2pDemoCommon {
         return data_size;
     }
 
+    static void dump_server_info(const server_info_t& server_info,
+                                 UcxLog &log)
+    {
+        log << server_info.conn->get_log_prefix()
+            << " read " << server_info.num_completed[IO_READ] << "/"
+            << server_info.num_sent[IO_READ] << " write "
+            << server_info.num_completed[IO_WRITE] << "/"
+            << server_info.num_sent[IO_WRITE];
+
+        if (server_info.conn->is_disconnecting()) {
+            log << " (disconnecting)";
+        }
+    }
+
+    void dump_timeout_waiting_for_replies_info()
+    {
+        size_t total_uncompleted = 0;
+        UcxLog log(LOG_PREFIX);
+
+        log << "timeout waiting for " << (_num_sent - _num_completed)
+            << " replies on the following connections:";
+
+        for (size_t i = 0; i < _active_servers.size(); ++i) {
+            if (get_num_uncompleted(_active_servers[i]) == 0) {
+                continue;
+            }
+
+            log << "\n";
```

Comment:
maybe move this "\n" to line 1432?

Reply:
we need to have "\n" per every `dump_server_info()`